frontend/__tests__/context/PlaygroundReducer.test.tsx

```diff
@@ -0,0 +1,540 @@
+      const initialState = createTestState();
+      const newDocument = {
+        id: "test-id",
+        name: "Test Document",
+        content: "Test content",
+      };
+
+      const action = {
+        type: "ADD_INPUT_DOCUMENT" as const,
+        payload: newDocument,
+      };
+      const newState = playgroundReducer(initialState, action);
+
+      expect(newState.inputDocuments).toHaveLength(1);
+      expect(newState.inputDocuments[0]).toEqual(newDocument);
+    });
+  });
+
+  describe("UPDATE_INPUT_DOCUMENT", () => {
+    it("should update an existing input document", () => {
+      const existingDocument = {
+        id: "test-id",
+        name: "Original Name",
+        content: "Original content",
+      };
+      const initialState = createTestState({
+        inputDocuments: [existingDocument],
+      });
+
+      const updatedDocument = {
+        id: "test-id",
+        name: "Updated Name",
+        content: "Updated content",
+      };
+      const action = {
+        type: "UPDATE_INPUT_DOCUMENT" as const,
+        payload: updatedDocument,
+      };
+      const newState = playgroundReducer(initialState, action);
+
+      expect(newState.inputDocuments).toHaveLength(1);
+      expect(newState.inputDocuments[0]).toEqual(updatedDocument);
+    });
+
+    it("should not modify other documents when updating one", () => {
+      const document1 = {
+        id: "id-1",
+        name: "Document 1",
+        content: "Content 1",
+      };
+      const document2 = {
+        id: "id-2",
+        name: "Document 2",
+        content: "Content 2",
+      };
+      const initialState = createTestState({
+        inputDocuments: [document1, document2],
+      });
+
+      const updatedDocument = {
+        id: "id-1",
+        name: "Updated Document 1",
+        content: "Updated Content 1",
+      };
+      const action = {
+        type: "UPDATE_INPUT_DOCUMENT" as const,
+        payload: updatedDocument,
+      };
+      const newState = playgroundReducer(initialState, action);
+
+      expect(newState.inputDocuments).toHaveLength(2);
+      expect(newState.inputDocuments[0]).toEqual(updatedDocument);
+      expect(newState.inputDocuments[1]).toEqual(document2);
+    });
+  });
+
+  describe("REMOVE_INPUT_DOCUMENT", () => {
+    it("should remove an input document by id", () => {
+      const document1 = {
+        id: "id-1",
+        name: "Document 1",
+        content: "Content 1",
+      };
+      const document2 = {
+        id: "id-2",
+        name: "Document 2",
+        content: "Content 2",
+      };
+      const initialState = createTestState({
+        inputDocuments: [document1, document2],
+      });
+
+      const action = {
+        type: "REMOVE_INPUT_DOCUMENT" as const,
+        payload: "id-1",
+      };
+      const newState = playgroundReducer(initialState, action);
+
+      expect(newState.inputDocuments).toHaveLength(1);
+      expect(newState.inputDocuments[0]).toEqual(document2);
```

Comment:
Looks like the code in this case is expecting the output docs to be cleared and it isn't tested here.  What are the rules around this?  It seems like any modification to transform invalidates the output docs, shouldn't that be true for the input docs too?

Reply:
The next test (line 121) is specifically on that behavior. Is that sufficient?